lib/dune_file.ml

```diff
@@ -198,26 +202,32 @@ module Packages = struct
         { changed; stanzas = List sexps; renames }
 
   let update_references renames sexps =
-    let changed, sexps =
-      List.fold_left
-        ~f:(fun (changed_before, acc) sexp ->
-          let { changed; stanzas; renames = _ } =
-            update_reference renames sexp
-          in
-          (changed_before || changed, stanzas :: acc))
-        ~init:(false, []) sexps
-    in
-    let sexps = List.rev sexps in
-    { changed; stanzas = sexps; renames }
+    let changed = false in
+    match sexps with
+    | [] -> { changed; stanzas = []; renames }
```

Comment:
I think the empty list case here is redundant as it's handled by the fold in the non-tuareg case below